pandas/io/packers.py

```diff
@@ -139,8 +139,12 @@ def writer(fh):
 
     path_or_buf = _stringify_path(path_or_buf)
     if isinstance(path_or_buf, str):
-        with open(path_or_buf, mode) as fh:
-            writer(fh)
+        try:
+            with open(path_or_buf, mode) as fh:
+                writer(fh)
+        except FileNotFoundError as error:
+            error.strerror = "File b'{}' does not exist".format(path_or_buf)
```

Comment:
hmm just do a 

```raise FileNotFoundError(msg)```

its prob ok generally to mutate the exception object, but we don't do this in pandas, so for consistency pls change.